### PR TITLE
Add support unify and split for content-item list

### PR DESCRIPTION
**Pull request by israelpoli**
## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-8809

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 * Fixed an issue where **validate** command failed on valid complex layout rules and triggers.
 * Fixed an issue where *validate* command failed with release notes files for assets modeling rules folder.
 * Added support for `skip` property in **pre-commit** hooks.
+* Added support of **unify** and **split** commands for List.
```
Comment (GuyAfik) [inline CHANGELOG.md:10]:
```suggestion
* Added support of the `List` content-item to the **prepare-content** and **split** commands.
```

File: demisto_sdk/commands/content_graph/objects/list.py [hunk 2]
```diff
@@ -31,3 +32,17 @@ def _upload(
...
+        **kwargs
+    ) -> dict:
+        data = (
+            self.data
+            if kwargs.get("unify_only")
+            else super().prepare_for_upload(current_marketplace)
+        )
+        data = ListUnifier.unify(self.path, self.data, marketplace=current_marketplace)
+
+        return data
```
Comment (GuyAfik) [inline demisto_sdk/commands/content_graph/objects/list.py:48]:
```suggestion
        return ListUnifier.unify(self.path, self.data, marketplace=current_marketplace)
```

File: demisto_sdk/commands/download/downloader.py [hunk 3]
```diff
@@ -1393,6 +1396,24 @@ def download_new_content_items(
                 if file_path.is_file():
                     downloaded_files.append(file_path)
 
+        elif content_item_entity_directory == LISTS_DIR:
```
Comment (GuyAfik) [inline demisto_sdk/commands/download/downloader.py:1399]: what is this change in the download command? do we need to document it too in the changelog?

File: demisto_sdk/commands/prepare_content/list_unifier.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+                "data section is not empty in "
+                f"{file_content_data.with_name(f'{file_content_data.parent.name}.json')} file. "
+                "It should be blank or a dash(-)."
+            )
+        json_unified["data"] = file_content_data.read_text()
+
+        return json_unified
+
+    @staticmethod
+    def find_file_content_data(list_path: Path) -> Path | None:
```
Comment (GuyAfik) [inline demisto_sdk/commands/prepare_content/list_unifier.py:45]: looks general method, doesn't make sense it would be under `ListUnifier` class

File: demisto_sdk/commands/split/tests/jsonsplitter_test.py [hunk 5]
```diff
@@ -48,3 +55,49 @@ def test_split_json(repo):
...
+    with ChangeCWD(pack.repo_path):
+        res = json_splitter.split_json()
+        assert res == 0
+        assert expected_list_path.is_file()
+        assert (
+            (expected_list_path.parent / f"{folder_and_file_name}_data")
+            .with_suffix(suffix)
+            .is_file()
+        )
+        assert get_dict_from_file(str(expected_list_path))[0]["data"] == "-"
```
Comment (GuyAfik) [inline demisto_sdk/commands/split/tests/jsonsplitter_test.py:103]: use the new file service, let me know how it goes :)

**Review by GuyAfik — COMMENTED** [review]
@israelpoli awesome job

File: demisto_sdk/commands/split/tests/jsonsplitter_test.py (hunk 5 above)
Comment (GuyAfik) [reply demisto_sdk/commands/split/tests/jsonsplitter_test.py:103]: https://github.com/demisto/demisto-sdk/tree/master/demisto_sdk/commands/common/files

File: demisto_sdk/commands/download/downloader.py (hunk 3 above)
Comment (israelpoli) [reply demisto_sdk/commands/download/downloader.py:1405]: This is definitely a change in **_download_** for `content-type list`
from now on during the download of a list it will be split

I will add it as a breaking change to the `CHANGELOG`

File: demisto_sdk/commands/download/downloader.py [hunk 6]
```diff
@@ -1435,7 +1462,10 @@ def download_existing_content_items(
         source_to_destination_mapping: dict[
             Path, Path
         ] = {}  # A mapping of temp file paths to target final paths
-        is_unified = content_item_entity_directory in (INTEGRATIONS_DIR, SCRIPTS_DIR)
+        is_unified = content_item_entity_directory in (INTEGRATIONS_DIR, SCRIPTS_DIR, LISTS_DIR)
+
+        if content_item_type == FileType.LISTS:
+            return self.download_new_content_items(content_object, output_path)
```
Comment (MichaelYochpaz) [inline demisto_sdk/commands/download/downloader.py:1468]: I think it's better and more readable to handle this on the upper function - `write_files_into_output_path` - instead of here, by changing:
```python
content_item_name: str = content_object["name"]
content_item_entity: str = content_object["entity"]
```
```python
if content_item_exists:
    downloaded_files = self.download_existing_content_items(
        content_object=content_object,
        existing_pack_structure=existing_pack_structure,
        output_path=output_path,
        temp_dir=temp_dir,
    )

else:
    downloaded_files = self.download_new_content_items(
        content_object=content_object,
        output_path=output_path,
    )
```

To:
```python
content_item_name: str = content_object["name"]
content_item_entity: str = content_object["entity"]
content_item_type: FileType = content_object["type"]
```
```python
# We skip 'download_existing_content_items' logic for lists since 'smart-merge' is irrelevant for lists
if content_item_exists and content_item_type != FileType.LISTS:
    downloaded_files = self.download_existing_content_items(
        content_object=content_object,
        existing_pack_structure=existing_pack_structure,
        output_path=output_path,
        temp_dir=temp_dir,
    )

else:
    downloaded_files = self.download_new_content_items(
        content_object=content_object,
        output_path=output_path,
    )
```

(The note there explaining why we ignore it is important for future reference)

**Review by MichaelYochpaz — COMMENTED** [review]
Looks good from my side of the review (`download.py`). Good job!
Left a small note.

Not approving yet since I want to test it on a live instance following the changes and assure everything works properly in some edge cases (like if the list was already downloaded, and is re-downloaded to the same path overriding existing files, for example).

File: CHANGELOG.md [hunk 7]
```diff
@@ -7,6 +7,7 @@
 * Fixed an issue where **validate** command failed on valid complex layout rules and triggers.
 * Fixed an issue where *validate* command failed with release notes files for assets modeling rules folder.
 * Added support for `skip` property in **pre-commit** hooks.
+* Added support of the `List` content-item to the **prepare-content** and **split** commands.
```
Comment (yuvalbenshalom) [inline CHANGELOG.md:10]: please document changes to upload in the change log.

File: demisto_sdk/commands/prepare_content/prepare_upload_manager.py [hunk 8]
```diff
@@ -2,7 +2,12 @@
 from pathlib import Path
 from typing import Optional, Union
 
-from demisto_sdk.commands.common.constants import MarketplaceVersions
+from demisto_sdk.commands.common.constants import (
+    JSON_INDENT_CONSTANT,
+    YAML_INDENT_CONSTANT,
```
Comment (yuvalbenshalom) [inline demisto_sdk/commands/prepare_content/prepare_upload_manager.py:7]: please explain to me later why they is relevant.

File: demisto_sdk/commands/split/jsonsplitter.py [hunk 9]
```diff
@@ -124,4 +145,56 @@ def create_module(self):
...
+        return output, file_name, file_data_name
+
+    def write_file_data(self, list_name_dir: Path, file_data_name: str):
+        if file_data_name.endswith("json"):
+            JsonFile.write_file(self.json_data["data"], (list_name_dir / file_data_name), indent=JSON_INDENT_CONSTANT)
+        else:
+            TextFile.write_file(self.json_data["data"], (list_name_dir / file_data_name))
+
+    def write_list(self, list_name_dir: Path, file_name: str):
+        self.json_data["data"] = "-"
```
Comment (yuvalbenshalom) [inline demisto_sdk/commands/split/jsonsplitter.py:192]: why do you put "-" in data?

**Review by yuvalbenshalom — CHANGES_REQUESTED** [review]
looks good. see my comments.

File: demisto_sdk/commands/prepare_content/list_unifier.py [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+                f"{file_content_data_path.with_name(f'{file_content_data_path.parent.name}.json')} file. "
+                "It should be blank or a dash(-)."
+            )
+        json_unified["data"] = TextFile.read_from_local_path(file_content_data_path)
+
+        return json_unified
+
+    @staticmethod
+    def find_file_content_data(list_path: Path) -> Optional[Path]:
+        for suffix in (".txt", ".json", ".html", ".css", ".csv", ".md"):
```
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/list_unifier.py:48]: Could there be more than one of these files in a folder?

File: demisto_sdk/commands/prepare_content/tests/list_unifier_test.py [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+    '''
+    pack = git_repo.create_pack("PackName")
+    list_json, list_data, list_dir_name = create_split_list()
+    list_dir_path = Path(pack.path) / list_dir_name
+    list_dir_path.mkdir(parents=True, exist_ok=True)
+    (list_dir_path / (list_dir_name.name + ".json")).write_text(str(list_json))
+    (list_dir_path / (list_dir_name.name + "_data.css")).write_text(str(list_data))
+
+    with ChangeCWD(pack.repo_path):
+        list_unifier = ListUnifier.unify(
```
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/tests/list_unifier_test.py:39]: does the Unifier care what the CWD is? 🤔 If so, why?

File: demisto_sdk/commands/prepare_content/list_unifier.py [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+            )
+            return data
+
+        json_unified = copy.deepcopy(data)
+        json_unified = ListUnifier.insert_data_to_json(json_unified, file_content_data_path)
+        logger.debug(f"[green]Created unified json: {path.name}[/green]")
+        return json_unified
+
+    @staticmethod
+    def insert_data_to_json(json_unified: Dict, file_content_data_path: Path):
```
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/list_unifier.py:35]: why do we need to insert data into a json?

File: demisto_sdk/commands/prepare_content/tests/list_unifier_test.py [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+    When:
+        - Running `find_file_content_data` method.
+    Then:
+        - Ensure the path of the data file is returned.
+    '''
+    (Path(tmpdir) / f"_data.{suffix}").write_text("data")
+    assert (
+        ListUnifier.find_file_content_data(Path(tmpdir) / "_data")
+        == Path(tmpdir) / f"_data.{suffix}"
+    )
```
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/tests/list_unifier_test.py:59]:
```suggestion
    path = Path(tmpdir, f"_data.{suffix}")
    path.write_text("data")
    assert (
        ListUnifier.find_file_content_data(path) == path
    )
```

reuse, reduce, recycle

File: demisto_sdk/commands/prepare_content/tests/list_unifier_test.py
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/tests/list_unifier_test.py:1]: missing type hints

File: demisto_sdk/commands/prepare_content/tests/list_unifier_test.py [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+    '''
+    Given:
+        - A data file with a suffix that is not txt, json, css, md, csv or html.
+    When:
+        - Running `find_file_content_data` method.
+    Then:
+        - Ensure None is returned.
+    '''
+    (Path(tmpdir) / f"_data.{suffix}").write_bytes(b"data")
+    assert not ListUnifier.find_file_content_data(Path(tmpdir) / "_data")
```
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/tests/list_unifier_test.py:73]:
```suggestion
    (path:= Path(tmpdir) / f"_data.{suffix}").write_bytes(b"data")
    assert not ListUnifier.find_file_content_data(path)
```

File: demisto_sdk/commands/split/jsonsplitter.py [hunk 15]
```diff
@@ -28,27 +37,40 @@ class JsonSplitter:
...
-        output: str,
+        input: Union[Path, str],
+        output: Union[Path, str] = "",
+        file_type: FileType = FileType.GENERIC_MODULE,
         no_auto_create_dir: bool = False,
         new_module_file: bool = False,
+        input_file_data: Optional[Dict] = None,
     ):
         self.input = input
+        self.output = output if output else Path(input).parent
```
Comment (dorschw) [inline demisto_sdk/commands/split/jsonsplitter.py:48]:
```suggestion
        self.output = output or Path(input).parent
```

File: demisto_sdk/commands/split/jsonsplitter.py [hunk 16]
```diff
@@ -124,4 +146,65 @@ def create_module(self):
 
             module_file_path = os.path.join(self.module_dir, file_name)
 
-        write_dict(module_file_path, data=self.module_json_data, indent=4)
+        write_dict(module_file_path, data=self.json_data, indent=JSON_INDENT_CONSTANT)
+
+    def get_auto_output_path(self) -> Tuple[Path, str, str]:
```
Comment (dorschw) [inline demisto_sdk/commands/split/jsonsplitter.py:151]: document

File: demisto_sdk/commands/split/jsonsplitter.py [hunk 17]
```diff
@@ -124,4 +146,65 @@ def create_module(self):
...
+        file_data_name = file_name[: -len(".json")] + "_data" + suffix
+
+        if self.autocreate_dir:
+            pack_name = get_pack_name(self.input)
+
+            if not pack_name:
+                return Path(self.input).parent, file_name, file_data_name
+
+            if not (lists_dir := Path(PACKS_DIR) / pack_name / LISTS_DIR).exists():
+                lists_dir.mkdir()
```
Comment (dorschw) [inline demisto_sdk/commands/split/jsonsplitter.py:172]:
```suggestion
            lists_dir := Path(PACKS_DIR) / pack_name / LISTS_DIR).mkdir(exists_ok=True)
```

File: demisto_sdk/commands/split/jsonsplitter.py [hunk 18]
```diff
@@ -124,4 +146,65 @@ def create_module(self):
...
+
+            # create the specific list dir under the LISTS_DIR if it does not exist
+            # the dir name determine by the file name without the suffix
+            if not (list_name_dir := lists_dir / file_name[: -len(".json")]).exists():
```
Comment (dorschw) [inline demisto_sdk/commands/split/jsonsplitter.py:176]: use `stem` to get the name without suffix.

Comment (dorschw) [inline demisto_sdk/commands/split/jsonsplitter.py:176]: apply the same `exists_ok=True` trick

File: demisto_sdk/commands/split/jsonsplitter.py [hunk 19]
```diff
@@ -124,4 +146,65 @@ def create_module(self):
...
+                list_name_dir.mkdir()
+
+            return list_name_dir, file_name, file_data_name
+
+        if not (output := Path(self.output)).is_dir():
+            output = output.parent
+
+        return output, file_name, file_data_name
+
+    def write_file_data(self, list_name_dir: Path, file_data_name: str):
```
Comment (dorschw) [inline demisto_sdk/commands/split/jsonsplitter.py:186]: Why does this method belong under a `JsonSplitter`?

File: demisto_sdk/commands/split/jsonsplitter.py [hunk 20]
```diff
@@ -124,4 +146,65 @@ def create_module(self):
...
+                json.loads(self.json_data["data"]),
+                (list_name_dir / file_data_name),
+                indent=JSON_INDENT_CONSTANT,
+            )
+        else:
+            TextFile.write_file(
+                self.json_data["data"], (list_name_dir / file_data_name)
+            )
+
+    def write_list(self, list_name_dir: Path, file_name: str):
```
Comment (dorschw) [inline demisto_sdk/commands/split/jsonsplitter.py:199]: Why does this method belong under a JsonSplitter?

Comment (dorschw) [inline demisto_sdk/commands/split/jsonsplitter.py:199]: why pass the file name and path separately?

File: demisto_sdk/tests/test_files/validate_integration_test_valid_types.py [hunk 21]
```diff
@@ -310,6 +310,32 @@
     "version": -1
 }
 
+LIST = {
```
Comment (dorschw) [inline demisto_sdk/tests/test_files/validate_integration_test_valid_types.py:313]: Confusing constant name, please either replace **or** use `as` when importing, to help maintainers what this mysterious constant is.

File: demisto_sdk/commands/split/tests/jsonsplitter_test.py [hunk 22]
```diff
@@ -48,3 +55,49 @@ def test_split_json(repo):
             result_generic_module = json.load(f)
 
         assert result_generic_module == GENERIC_MODULE
+
+
+@pytest.mark.parametrize(
```
Comment (dorschw) [inline demisto_sdk/commands/split/tests/jsonsplitter_test.py:60]: nice

File: demisto_sdk/commands/split/tests/jsonsplitter_test.py [hunk 23]
```diff
@@ -48,3 +55,49 @@ def test_split_json(repo):
...
+    When:
+        - Running `JsonSplitter.split_json` on it.
+    Then:
+        - Ensure the specific list dir created.
+        - Ensure the data file of the list with the extension matched has been created.
+        - Ensure the data section in the json list file is filled with `-`.
+    """
+    pack = repo.create_pack("PackName")
+    list_json = pack.create_list("test-list", LIST)
+    list_json.update({"id": list_name, "name": list_name, "type": list_type, "data": data})
```
Comment (dorschw) [inline demisto_sdk/commands/split/tests/jsonsplitter_test.py:84]: why do we need this line?

**Review by dorschw — CHANGES_REQUESTED** [review]
Nice job, see the comments.

File: demisto_sdk/commands/download/downloader.py [hunk 24]
```diff
@@ -1128,6 +1131,12 @@ def get_metadata_file(content_type: str, content_item_path: Path) -> dict | None
                 return get_yaml(content_item_path)
 
         else:
+            if content_type == LISTS_DIR:
+                if content_item_path.is_dir():
+                    json_files = [str(path) for path in content_item_path.iterdir() if path.suffix == ".json"]
+                    if not json_files:
+                        return None
+                    return get_json(json_files[0])
```
[the remark this answers is not in the document]
Comment (israelpoli) [reply demisto_sdk/commands/download/downloader.py:1139]: There can't be another `json` unless it's a list of type `json`, in which case I've added special handling

File: demisto_sdk/commands/download/downloader.py [hunk 25]
```diff
@@ -1274,6 +1283,7 @@ def write_files_into_output_path(
             for file_name, content_object in downloaded_content_objects.items():
                 content_item_name: str = content_object["name"]
                 content_item_entity: str = content_object["entity"]
+                content_item_type: FileType = content_object["type"]
```
[the remark this answers is not in the document]
Comment (israelpoli) [reply demisto_sdk/commands/download/downloader.py:1286]: All content objects that is in `downloaded_content_objects` must be with a type

File: demisto_sdk/commands/common/constants.py [hunk 26]
```diff
@@ -1963,3 +1963,15 @@ class IncidentState(str, Enum):
...
+class TypeList(str, Enum):
+    TEXT = "plain_text"
+    HTML = "html"
+    CSS = "css"
+    MD = "markdown"
+    JSON = "json"
+
+# Used to format the writing of the yml/json file
+JSON_INDENT_CONSTANT = 4
+YAML_INDENT_CONSTANT = 0
```
[the remark this answers is not in the document]
Comment (israelpoli) [reply demisto_sdk/commands/common/constants.py:1977]: Wherever you write `content-item`, if it is json, then the indent=4, if it is yml, then the indent=0 that's why I created constants (I also use it several times throughout the feature)

File: demisto_sdk/commands/prepare_content/list_unifier.py [hunk 27]
```diff
@@ -0,0 +1,51 @@
...
+        if json_unified.get("data", "") not in ("", "-"):
+            logger.warning(
+                "data section is not empty in "
+                f"{file_content_data_path.with_name(f'{file_content_data_path.parent.name}.json')} file. "
+                "It should be blank or a dash(-)."
+            )
```
[the remark this answers is not in the document]
Comment (israelpoli) [reply demisto_sdk/commands/prepare_content/list_unifier.py:41]: I implemented this part similarly to handling scripts or integrations in unify

the warning should show the user that we have overridden the section `data` that was written in the list file
but on the other hand, do not issue an error because the user run the unify command

File: demisto_sdk/commands/prepare_content/list_unifier.py (hunk 10 above)
Comment (israelpoli) [reply demisto_sdk/commands/prepare_content/list_unifier.py:48]: no, it is a closed list

File: demisto_sdk/commands/split/jsonsplitter.py (hunk 9 above)
Comment (israelpoli) [reply demisto_sdk/commands/split/jsonsplitter.py:200]: Similar to the implementation of unify in scripts and integrations, we set the data/script to "-"
but we also allow to leave it empty

File: CHANGELOG.md (hunk 7 above)
Comment (israelpoli) [reply CHANGELOG.md:10]: I added a description of the `download`
Regarding the `upload`, it remains the same situation

File: demisto_sdk/commands/content_graph/parsers/json_content_item.py [hunk 28]
```diff
@@ -94,9 +95,14 @@ def marketplaces(self) -> List[MarketplaceVersions]:
     def get_json(self, git_sha: Optional[str]) -> Dict[str, Any]:
         if self.path.is_dir():
             json_files_in_dir = get_files_in_dir(self.path.as_posix(), ["json"], False)
-            if len(json_files_in_dir) != 1:
-                raise NotAContentItemException(
-                    f"Directory {self.path} must have a single JSON file."
-                )
+            if len(json_files_in_dir) > 1:
```
[the remark this answers is not in the document]
Comment (israelpoli) [reply demisto_sdk/commands/content_graph/parsers/json_content_item.py]: I rewrote the logic here

When there is more than one json file it will raise an error except for the case that the list type is json and then I ignore the *_data.json file

File: demisto_sdk/commands/prepare_content/prepare_upload_manager.py (hunk 8 above)
Comment (israelpoli) [reply demisto_sdk/commands/prepare_content/prepare_upload_manager.py:7]: because every time I write json/yml I need to redefine `indent` I created constants for this

File: demisto_sdk/commands/split/jsonsplitter.py (hunk 19 above)
Comment (israelpoli) [reply demisto_sdk/commands/split/jsonsplitter.py:188]: According to ymlspliter where you also write to the file, I implemented the same functionality here as well, it is not similar to `unify` where we have a parent `PrepareUploadManager`

File: demisto_sdk/commands/split/jsonsplitter.py (hunk 20 above)
Comment (israelpoli) [reply demisto_sdk/commands/split/jsonsplitter.py]: According to ymlspliter where you also write to the file, I implemented the same functionality here as well, it is not similar to `unify` where we have a parent `PrepareUploadManager`

File: demisto_sdk/commands/prepare_content/list_unifier.py (hunk 12 above)
Comment (israelpoli) [reply demisto_sdk/commands/prepare_content/list_unifier.py:36]: Similar to other unify, I also implement this method here, that is, both to unify the data with the list and to issue a warning when the list file has values in the data section

File: demisto_sdk/commands/split/jsonsplitter.py [hunk 29]
```diff
@@ -124,4 +146,69 @@ def create_module(self):
...
+            TypeListData.HTML: ".html",
+            TypeListData.CSS: ".css",
+            TypeListData.MD: ".md",
+            TypeListData.JSON: ".json",
+        }
+
+        suffix = suffix_by_type.get(self.json_data["type"], ".txt")
+
+        file_name = pascal_case(self.json_data["name"]) + ".json"
+        file_data_name = file_name[: -len(".json")] + "_data" + suffix
```
Comment (dorschw) [inline demisto_sdk/commands/split/jsonsplitter.py:167]: using `Path` objects, `.with_stem` can simplify this.

File: demisto_sdk/commands/split/jsonsplitter.py [hunk 30]
```diff
@@ -124,4 +146,69 @@ def create_module(self):
...
+            pack_name = get_pack_name(self.input)
+
+            if not pack_name:
+                return Path(self.input).parent, file_name, file_data_name
+
+            (lists_dir := Path(PACKS_DIR) / pack_name / LISTS_DIR).mkdir(exist_ok=True)
+
+            # create the specific list dir under the LISTS_DIR if it does not exist
+            # the dir name determine by the file name without the suffix
+            (list_name_dir := lists_dir / Path(file_name).stem).mkdir(exist_ok=True)
```
Comment (dorschw) [inline demisto_sdk/commands/split/jsonsplitter.py:179]:
```suggestion
            (list_name_dir := Path(PACKS_DIR, pack_name, LISTS_DIR, Path(file_name).stem).mkdir(parents=True, exist_ok=True)
```

1. Removed the comment since it explains what is done, instead of why
2. Using `parents=True` saves the need to create the two dirs separately.

File: demisto_sdk/commands/split/jsonsplitter.py [hunk 31]
```diff
@@ -124,4 +146,69 @@ def create_module(self):
...
+        if file_data_name.endswith("json"):
+            try:
+                JsonFile.write_file(
+                    json.loads(self.json_data["data"]),
+                    (list_name_dir / file_data_name),
+                    indent=DEFAULT_JSON_INDENT,
+                )
+            except json.decoder.JSONDecodeError as e:
+                raise Exception(
+                    f"Could not parse data of the list {self.json_data['name']}.\n"
```
Comment (dorschw) [inline demisto_sdk/commands/split/jsonsplitter.py:198]: not necessarily a list

Comment (dorschw) [inline demisto_sdk/commands/split/jsonsplitter.py:198]: what if there's no `'name'` in `self.json_data`?

Comment (dorschw) [inline demisto_sdk/commands/split/jsonsplitter.py:198]: I'd split the reading from writing. The `try` is on the writing but excepts an error in parsing (reading), which is confusing.

File: demisto_sdk/commands/split/jsonsplitter.py (hunk 9 above)
Comment (dorschw) [reply demisto_sdk/commands/split/jsonsplitter.py:206]: Please add a comment mentioning this

File: demisto_sdk/commands/split/jsonsplitter.py [hunk 32]
```diff
@@ -124,4 +146,69 @@ def create_module(self):
 
             module_file_path = os.path.join(self.module_dir, file_name)
 
-        write_dict(module_file_path, data=self.module_json_data, indent=4)
+        write_dict(module_file_path, data=self.json_data, indent=DEFAULT_JSON_INDENT)
+
+    def get_auto_output_path(self) -> Tuple[Path, str, str]:
```
Comment (dorschw) [inline demisto_sdk/commands/split/jsonsplitter.py:151]: We can simplify it and return just one object (the file path), since the folder and the data path can be calculated from it, **and** this is only used once.

Comment (dorschw) [inline demisto_sdk/commands/split/jsonsplitter.py:151]: does `get_auto_output_path` apply only to lists, or to other content items as well? If it's only for lists, let's mention it in the method name.

File: demisto_sdk/commands/split/jsonsplitter.py (hunk 31 above)
Comment (israelpoli) [reply demisto_sdk/commands/split/jsonsplitter.py:198]: A list object must have a name field with a value

Comment (israelpoli) [reply demisto_sdk/commands/split/jsonsplitter.py:198]: this code is only for list content-item

**Review by dorschw — APPROVED** [review]

**Comment by tschanfeld** [comment]
@GuyAfik i don't think this affects the current documentation
